### PR TITLE
Consolidate OS dependant functions.

### PR DESCRIPTION
**Pull request by andrewhood125**
@taylorotwell @svpernova09

### CODE REVIEW

File: src/Support/Portability.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+     * Operating System Is OSX?
+     *
+     * @return boolean
+     */
+    private static function isMac()
+    {
+        return strpos(strtoupper(PHP_OS), 'DARWIN') === 0;
+    }
+
+
```
Comment (GrahamCampbell) [inline src/Support/Portability.php]: cs

File: src/Support/Portability.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+     */
+    private static function isWindows()
+    {
+        return strpos(strtoupper(PHP_OS), 'WIN') === 0;
+    }
+
+    /**
+     * Operating System Is OSX?
+     *
+     * @return boolean
```
Comment (GrahamCampbell) [inline src/Support/Portability.php]: bool

File: src/Support/Portability.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        return 'VAGRANT_DOTFILE_PATH="'.$_ENV['VAGRANT_DOTFILE_PATH'].'"';
+    }
+
+
+    /**
+     * Operating System Is Windows?
+     *
+     * @return boolean
```
Comment (GrahamCampbell) [inline src/Support/Portability.php]: bool

File: src/Support/Portability.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+     */
+    private static function isWindows()
```
Comment (GrahamCampbell) [inline src/Support/Portability.php]: must be protected unless you make the class final

File: src/Support/Portability.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        return strpos(strtoupper(PHP_OS), 'WIN') === 0;
+    }
+
+    /**
+     * Operating System Is OSX?
+     *
+     * @return boolean
+     */
+    private static function isMac()
```
Comment (GrahamCampbell) [inline src/Support/Portability.php]: must be protected unless you make the class final

File: src/Support/Portability.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    public static function setEnvironmentCommand()
+    {
+        if (self::isWindows()) {
+            return 'SET VAGRANT_DOTFILE_PATH='.$_ENV['VAGRANT_DOTFILE_PATH'].' &&';
+        }
+
+        return 'VAGRANT_DOTFILE_PATH="'.$_ENV['VAGRANT_DOTFILE_PATH'].'"';
+    }
+
+
```
Comment (GrahamCampbell) [inline src/Support/Portability.php]: cs

File: src/Support/Portability.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            return 'start';
+        } elseif (self::isMac()) {
+            return 'open';
+        }
+
+        return 'xdg-open';
+    }
+
+    /**
+     * Set Vagrant Dotfile Path
```
Comment (GrahamCampbell) [inline src/Support/Portability.php]: cs

Comment (andrewhood125) [inline src/Support/Portability.php]: Like this `Set vagrant dotfile path`?

**Comment by andrewhood125** [comment]
Thanks @GrahamCampbell. Updated.

File: src/Support/Portability.php (hunk 7 above)
Comment (GrahamCampbell) [inline src/Support/Portability.php]: You need a fullstop.

Comment (andrewhood125) [inline src/Support/Portability.php]: Ah gotcha. Updated.

**Comment by taylorotwell** [comment]
There are a lot of CS problems I don't really have time to fix at the moment. I will try to list the ones I see off hand:

- DocBlocks don't match rest of framework.
- You have spacing in between concatenation operators.
- Imports are organized by length.

**Comment by andrewhood125** [comment]
If this is something you'd like to see, I can clean it up. What say you?

**Comment by andrewhood125** [comment]
@taylorotwell ^

**Pull request closed without merging**